gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/iapi/sql/dictionary/StatementTablePermission.java

```diff
@@ -134,7 +135,9 @@ public void check( LanguageConnectionContext lcc,
 	{
 		DataDictionary dd = lcc.getDataDictionary();
 	
-		if( ! hasPermissionOnTable( dd, authorizationId, forGrant))
+		if( ! hasPermissionOnTable( dd, authorizationId, forGrant) && !(getTableDescriptor(dd)
+				.getSchemaName().equalsIgnoreCase(Misc.SNAPPY_HIVE_METASTORE) && this.privType ==
```

Comment:
Indentation of the second line needs to be corrected. Should just start after 2 spaces from the above line.